File: 855_Maximum_Distance_Between_a_Pair_of_Values/solution.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        i, j, a = 0, 0, 0
        while i < len(nums1) and j < len(nums2):
            if nums1[i] <= nums2[j]:
                a = max(a, j - i)
                j += 1
            else:
                i += 1
                if i > j: j = i
        return a

File: 855_Maximum_Distance_Between_a_Pair_of_Values/solution.c
int maxDistance(int* nums1, int n1s, int* nums2, int n2s) {
    int i = 0, j = 0, a = 0;
    while (i < n1s && j < n2s) {
        if (nums1[i] <= nums2[j]) {
            a = (a > j - i) ? a : j - i;
            j++;
        } else {
            i++;
            if (i > j) j = i;
        }
    }
    return a;
}
